chain/stmgr/searchwait.go

```diff
@@ -159,6 +173,21 @@ func (sm *StateManager) SearchForMessage(ctx context.Context, head *types.TipSet
 	return fts, r, foundMsg, nil
 }
 
+func (sm *StateManager) searchForIndexedMsg(ctx context.Context, mcid cid.Cid, m types.ChainMsg) (*types.TipSet, *types.MessageReceipt, cid.Cid, error) {
+	minfo, err := sm.msgIndex.GetMsgInfo(ctx, mcid)
+	if err != nil {
+		return nil, nil, cid.Undef, err
+	}
+
+	ts, err := sm.cs.GetTipSetByCid(ctx, minfo.TipSet)
+	if err != nil {
+		return nil, nil, cid.Undef, err
+	}
+
+	r, foundMsg, err := sm.tipsetExecutedMessage(ctx, ts, mcid, m.VMMessage(), false)
+	return ts, r, foundMsg, err
+}
```

Comment:
We should use the execution tipset and the execution index to just rescue the receipt and call it a day here, instead of handing off to `tipsetExecutedMessage`. However, @arajasek wanted to contain the changeset leaving out the execution index from the index, in which case this logic can stay.

Reply:
we dropped the index from the index.